.github/workflows/spark-ci.yml

```diff
@@ -69,7 +69,7 @@ jobs:
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        jvm: [8, 11]
+        jvm: [8, 11, 17]
```

Comment:
> 17 exercised by spark-3x-java-17-tests

it works for both 2.12 and 2.13, if you want we can merge the last one into this single runs
- https://github.com/apache/iceberg/pull/7391

Reply:
i think the person writing those jobs didn't know about matrix excludes/includes (or didn't want to use them).
we should be able to merge the jobs and remove duplication. 
would prefer for such changes to be considered as a follow-up though

Reply:
+1 on removing `spark-3x-java-17-tests` and including `17` into the matrix for testing Spark

Reply:
done in https://github.com/apache/iceberg/pull/10513.
changes in this file backed out for this PR

thank you @singhpk234 @nastra !

Reply:
looks like the changes in this file disappeared. It would actually be good to clean this up in the scope of this PR